sklearn/multioutput.py

```diff
@@ -0,0 +1,221 @@
+"""
+MultiOutputClassifier
+===========================
+This module includes several classes that extend base estimators to
+multi-target estimators.
+
+Most sklearn estimators use a response matrix to train a target function
+with a single output variable.  I.e. typical estimators use the
+training set X to estimate a target function f(X) that
+predicts a single y. The purpose of this class is to extend estimators
+to be able to estimate a series of target functions (f1,f2,f3...,fn)
+that are trained on a single X predictor matrix to predict a series
+of reponses (y1,y2,y3...,yn).
+"""
+
+# Author: Hugo Bowne-Anderson <hugobowne@gmail.com>
+# Author: Chris Rivera <chris.richard.rivera@gmail.com>
+# Author: Michael Williamson
+# Author: James Ashton Nichols <james.ashton.nichols@gmail.com>
+
+# License: BSD 3 clause
+
+import numpy as np
+import warnings
+import scipy.sparse as sp
+
+from .base import ClassifierMixin, MetaEstimatorMixin, BaseEstimator, clone
+from .preprocessing import LabelBinarizer
+from .metrics.pairwise import euclidean_distances
+from .utils import check_random_state, check_array
+from .utils.estimator_checks import multioutput_estimator_convert_y_2d
+from .utils.validation import _num_samples
+from .utils.validation import check_consistent_length
+from .utils.validation import check_is_fitted
+from .externals.joblib import Parallel, delayed
+from .externals import six
+
+from sklearn.multiclass import OneVsRestClassifier
+
+
+def _check_estimator(estimator):
+    """Make sure that an estimator implements the necessary methods."""
+    if (not hasattr(estimator, "score") and
+            not hasattr(estimator, "predict_proba")):
+        raise ValueError("The base estimator should implement "
+                         "score or predict_proba!")
+
+
+class MultiOutputClassifier(BaseEstimator, ClassifierMixin, MetaEstimatorMixin):
+    """
+    Converts a single target classifer estimator into
+    a multi-target classifier estimator.
+
+    This class fits and predicts a series of one-versus-all models
```

Comment:
series of one-versus-all models -> series of models . The base estimator need not do a one-vs-all prediction as always